src/test/java/com/box/sdk/MetadataTest.java

```diff
@@ -0,0 +1,119 @@
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+public class MetadataTest {
+
+    @Test
+    @Category(UnitTest.class)
+    public void testConstructor() {
+        Metadata m = new Metadata();
+        Assert.assertEquals("{}", m.toString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testCopyConstructor() {
+        Metadata m1 = new Metadata().add("/foo", "bar");
+        Metadata m2 = new Metadata(m1);
+        Assert.assertEquals("{\"foo\":\"bar\"}", m2.toString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testAdd() {
+        Metadata m = new Metadata().add("/foo", "bar");
+        JsonArray operations = JsonArray.readFrom(m.getPatch());
+        Assert.assertEquals(1, operations.size());
+        JsonObject op = operations.get(0).asObject();
+        Assert.assertEquals("add", op.get("op").asString());
+        Assert.assertEquals("/foo", op.get("path").asString());
+        Assert.assertEquals("bar", op.get("value").asString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testReplace() {
+        Metadata m = new Metadata().replace("/foo", "bar");
+        JsonArray operations = JsonArray.readFrom(m.getPatch());
+        Assert.assertEquals(1, operations.size());
+        JsonObject op = operations.get(0).asObject();
+        Assert.assertEquals("replace", op.get("op").asString());
+        Assert.assertEquals("/foo", op.get("path").asString());
+        Assert.assertEquals("bar", op.get("value").asString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testTest() {
+        Metadata m = new Metadata().test("/foo", "bar");
+        JsonArray operations = JsonArray.readFrom(m.getPatch());
+        Assert.assertEquals(1, operations.size());
+        JsonObject op = operations.get(0).asObject();
+        Assert.assertEquals("test", op.get("op").asString());
+        Assert.assertEquals("/foo", op.get("path").asString());
+        Assert.assertEquals("bar", op.get("value").asString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testRemove() {
+        Metadata m = new Metadata().remove("/foo");
+        JsonArray operations = JsonArray.readFrom(m.getPatch());
+        Assert.assertEquals(1, operations.size());
+        JsonObject op = operations.get(0).asObject();
+        Assert.assertEquals("remove", op.get("op").asString());
+        Assert.assertEquals("/foo", op.get("path").asString());
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testInvalidGet() {
+        Metadata m = new Metadata();
+        Assert.assertEquals(null, m.get("/foo"));
+    }
+
+    @Test
+    @Category(UnitTest.class)
+    public void testNullPath() {
+        try {
+            new Metadata().add(null, "value");
+            Assert.fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException ex) {
+        }
+    }
+
+    @Test
```

Comment:
Yeah...  Using "expected" causes problems for EclEmma (code coverage tool), so I try to avoid it.  If that's your preference, I'm happy to make the change.